src/main/java/com/nkh/ECommerceShop/security/service/RefreshTokenService.java

```diff
@@ -0,0 +1,60 @@
+package com.nkh.ECommerceShop.security.service;
+
+import com.nkh.ECommerceShop.exception.TokenRefreshException;
+import com.nkh.ECommerceShop.model.RefreshToken;
+import com.nkh.ECommerceShop.repository.RefreshTokenRepository;
+import com.nkh.ECommerceShop.repository.UsersRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class RefreshTokenService {
+    @Value("${app.auth.jwtRefreshExpirationMs}")
+    private Long refreshTokenDurationMs;
+
+    @Autowired
```

Comment:
can you remove @Autowired from fields since you have a construtor

Reply:
fixed